qa/tasks/cephfs/test_cephfs_shell.py

```diff
@@ -162,6 +162,124 @@ def test_mkdir_path_with_path_option(self):
         o = self.mount_a.stat('d5/d6/d7')
         log.info("mount_a output:\n{}".format(o))
 
+class TestRMdir(TestCephFSShell):
+    def test_rmdir(self):
+        """
+        Test that rmdir deletes directory
+        """
+        o = self.get_cephfs_shell_cmd_output("mkdir test_dir")
+        log.info("cephfs-shell output:\n{}".format(o))
+
+        o = self.get_cephfs_shell_cmd_output("rmdir test_dir")
+        log.info("cephfs-shell output:\n{}".format(o))
+
+        try:
+            o = self.mount_a.stat('test_dir')
+            log.info("mount_a output:\n{}".format(o))
+            raise Exception("Something went wrong!! test_dir is not deleted")
+        except CommandFailedError:
+            pass
+
+    def test_rmdir_empty_sub_dir(self):
+        """
+        Test that rmdir does not deletes empty child directories
+        """
+        o = self.get_cephfs_shell_cmd_output("mkdir -p test_dir/t1/t2/t3")
+        log.info("cephfs-shell output:\n{}".format(o))
+
+        o = self.run_cephfs_shell_cmd("rmdir test_dir")
+        print_output = o.stderr.getvalue()
+        log.info("cephfs-shell output:\n{}".format(print_output))
+        assert print_output.find('Directory not empty') != -1
```

Comment:
I would prefer checking for failure and `stat`ing for presence of `test_dir`. This way we would also test a bug where the method might delete the underlying file or the directory itself and yet somehow ends with valid failure.

Depending on error message is not the best, IMO, since minor change in it would make updating tests mandatory.

And if you do want match test, use case-insensitive search or setting everything to lowercase can prevent scope of irritation in future.

Reply:
> Depending on error message is not the best, IMO, since minor change in it would make updating tests mandatory.

Happened to me today, test_snap is broken -
```
CephFS:~/>>> snap create snap1 /data_dir
snapshot 'snap1' already exists
CephFS:~/>>>
```

From https://github.com/ceph/ceph/blob/master/qa/tasks/cephfs/test_cephfs_shell.py#L346 -
```
 assert(o[0] == "ERROR: snapshot 'snap1' already exists")
```

Reply:
You are right, I will change it.

Reply:
Cool.

Also, [uniittest provides a large list of assert methods](https://docs.python.org/3.7/library/unittest.html#assert-methods), use those instead of the `assert` keyword since it leads to a better output in case of a failure.